rust/capture/src/v0_endpoint.rs

```diff
@@ -182,9 +182,13 @@ pub fn process_single_event(
         return Err(CaptureError::MissingEventName);
     }
 
-    let data_type = match context.historical_migration {
-        true => DataType::AnalyticsHistorical,
-        false => DataType::AnalyticsMain,
+    let data_type = match event.event.as_str() {
```

Comment:
I don't have a _strong_ preferance about this, so feel free to ignore, but I'd generally prefer to write this like, avoiding the nested match:

```rust
let data_type = match (event.event.as_str(), context.historical_migration) {
    ("$exception", _) => DataType::ExceptionMain,
    ("$$heatmap", _) => DataType::HeatmapMain,
    (_, true) => DataType::AnalyticsHistorical,
    (_, false) => DataType::AnalyticsMain
};
```